setup.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+
+# If a binary distribution exists for your OS and Python, congratulations.  You should
+# be able to simply install swig.
+#
+#
+# If you are doing a build from sources, you must also have have swig and a copy of
+# the cspice toolkit.
+#
+# [I need appropriate instructions here on how to get cspice.a and install it. so that
+#    <this top level directory>/cspice/lib/libcspice.a
+#    <this top level directory>/cspice/include/
+# contains the library and and the include files]
+#
+# Execute the following:
+#     python setup.py build_ext --inplace
+#
+# To test the installation, the following should display the CSPICE
+# toolkit version string:
+#
+#    $ python
+#    >>> import cspyce
+#    >>> cspyce.tkvrsn("toolkit")
+#
+
+
+# We prefer setuptools, but will use distutils if setuptools isn't available
+
+try:
+    from setuptools import Command, setup, Extension
+    from setuptools.command.build_py import build_py
+except:
+    from distutils.core import Command, setup, Extension
+    from distutils.core import setup, Extension, build_py
+
+import numpy
+from glob import glob
+import subprocess
+import sys
+
+PYTHON2 = sys.version_info[0] < 3
+
+class GenerateCommand(Command):
+    description = 'Create generated files'
+    user_options = []
+
+    def initialize_options(self):
+        pass
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        if not PYTHON2:
+            print("Recreating the vectorize macros")
+            from swig.make_vectorize import create_vectorize_header_file
+            create_vectorize_header_file("swig/vectorize.i")
+            print("Rerunning swig")
+            command = "swig -python -outdir cspyce/. -o swig/cspyce0_wrap.c swig/cspyce0.i".split(' ')
+            subprocess.check_call(command)
+            command = "swig -python -outdir cspyce/. -o swig/typemap_samples_wrap.c swig/typemap_samples.i".split(' ')
+            subprocess.check_call(command)
+        else:
+            command = "echo Cannot rebuild files in Python2".split(' ')
+            subprocess.check_call(command)
+
+cspice_module = Extension(
+    'cspyce.cspice',
+    sources=glob("cspice/src/cspice/*.c"),
+    include_dirs=['cspice/include'],
+    extra_compile_args=['-Wno-incompatible-pointer-types', '-Wno-parentheses', 
```

Comment:
It's going to be really interesting to try this on Windows, where these compiler flags don't work.

Reply:
Yeah.  The cspice source code is sort of scary with all the the problems that the C compiler notices.  I wonder if the C compiler is uncovering some actual bugs.